### PR TITLE
fix #27267, print bytes of multi-byte characters at once on windows

### LINKED ISSUE #27267: display() problem printing multi-bytes characters
```
julia> versioninfo()
Julia Version 0.7.0-DEV.5219
Commit b65712285a* (2018-05-25 14:37 UTC)
Platform Info:
  OS: Windows (i686-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-3630QM CPU @ 2.40GHz
  WORD_SIZE: 32
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, ivybridge)

julia> display("δ"); println("δ")
"��"
δ

julia> display("한"); println("한")
"���"
한

julia> display("aδ한a"); println("aδ한a")
"a�����a"
aδ한a

julia>
```

### PR DESCRIPTION
**Pull request by JeffBezanson**
I would really like to release 0.7, so it's time for some undisciplined hacking. Hopefully this fixes the problem for now, replacing it with a performance problem. Writing output one byte at a time is not exactly speedy anyway, so maybe it's not too bad.

### CODE REVIEW

**Comment by alkorang** [comment]
I'm worried that it is really thread-safe. I mean, not for multithreading, but for asychronous I/O. Because `write(io::IO, c::Char)` is not only for `stdout`, so maybe if we use global variable and calling like :
```julia
@sync begin
    @async println('한'); # 3 bytes UTF-8
    @async write(txtfile, 'δ'); # 2 bytes UTF-8
end
```
I'm not sure this will not overwrite the global buffer and wrong values are written. (e.g write first two bytes of `'한'` into `txtfile` instead of `'δ'`) I think using local variable will be safer.

By the way, from my old pr #27380, some tests failed when writing into Windows process pipelines. I hope this can solve the problem.

File: base/io.jl [hunk 1]
```diff
@@ -564,7 +564,23 @@ function write(s::IO, a::SubArray{T,N,<:Array}) where {T,N}
...
+    if n == 1
+        return write(io, UInt8(v >>> 24))
+    end
+    # TODO: replace this with a Ref when stack copying is turned off, or with buffering inside libuv.
+    buf = fill(hton(v))
+    GC.@preserve buf begin
+        p = unsafe_convert(Ptr{Cvoid}, buf)
+        unsafe_write(io, p, n)
+    end
+    return n
```
Comment (vtjnash) [inline base/io.jl]: Should be able to replace all of this with `write(io, String(c))` (which presumably, should be implemented exactly like this)

**Review by vtjnash — COMMENTED** [review]
Wow, this seems like it was significantly more work then addressing it in libuv! Do we at least know why/when/which commit caused this to regress?

File: base/io.jl [hunk 2]
```diff
@@ -564,7 +564,23 @@ function write(s::IO, a::SubArray{T,N,<:Array}) where {T,N}
     end
 end
 
+const _os = ccall(:jl_get_UNAME, Any, ())
+
 function write(io::IO, c::Char)
+@static if _os === :Windows || _os === :NT
```
Comment (alkorang) [inline base/io.jl]: I think we don't need to seperate Windows and others. Using `Ref` worked well on the others, so I think using `fill` will also be okay.

File: base/io.jl (hunk 1 above)
Comment (JeffBezanson) [reply base/io.jl]: Good point, I guess we could add this as a fast implementation of `String(::Char)`.

File: base/io.jl (hunk 2 above)
Comment (alkorang) [reply base/io.jl]: How about replacing `while` with `unsafe_write` for other platforms by removing `if _os === :Windows || _os === :NT` statement. (Of course, if `unsafe_write` is faster.) Eventually this implementation may go into `string(::Char)` and we use `write(io::IO, c::Char) = write(io, string(c))`.

**Comment by JeffBezanson** [comment]
@vtjnash found the offending libuv change, so this shouldn't be necessary.

**Pull request closed without merging**